### PR TITLE
Enable product tests to be executed with different JDBC driver

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-product-tests/conf/docker/multinode-tls/docker-compose.yml [hunk 1]
```diff
@@ -45,5 +45,5 @@ services:
       - presto-master
 
   application-runner:
-    volumes:
-      - ../../../conf/tempto/tempto-configuration-for-docker-tls.yaml:/docker/volumes/tempto/tempto-configuration-local.yaml
+    environment:
```
Comment (losipiuk) [inline presto-product-tests/conf/docker/multinode-tls/docker-compose.yml]: Unrelated comment:

I find the structure of those docker-compose.yml files really hard to follow. The configuration is setup via a number of `compose.sh` files and it is hard to understand what is derived from what. It does not seem like having definition externally makes us be more "composition over inheritance" as we are still defining inheritance hierarchy, yet in external file.

IMO it would be nicer to have it combined so the relation between *.yml files is within files themselves. Is that possible?
What do you think?

File: presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+databases:
```
Comment (losipiuk) [inline presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml]: The `tempto-configuration-for-docker-kerberos-simba.yaml` and `tempto-configuration-for-docker-ldap-simba.yaml` are not referenced anywhere. Even if we do not have access to simba driver from OSS travis it would be nice to have a defnition in prestodb/master so as little as possible need to be redefined by user to run tests on simba drivers.

File: presto-product-tests/conf/docker/common/compose-commons.sh [hunk 3]
```diff
@@ -52,3 +52,11 @@ export HIVE_PROXY_PORT=${HIVE_PROXY_PORT:-1180}
...
 export LDAP_SERVER_HOST=${LDAP_SERVER_HOST:-doesntmatter}
 export LDAP_SERVER_IP=${LDAP_SERVER_IP:-127.0.1.1}
+
+if [[ -z ${PRESTO_JDBC_DRIVER_JAR} ]]; then
+    source "${PRODUCT_TESTS_ROOT}/target/classes/presto.env"
+    PRESTO_JDBC_DRIVER_JAR="${PROJECT_ROOT}/presto-jdbc/target/presto-jdbc-${PRESTO_VERSION}.jar"
+fi
+export_canonical_path PRESTO_JDBC_DRIVER_JAR
+
+export PRESTO_JDBC_DRIVER_CLASS=${PRESTO_JDBC_DRIVER_CLASS:-"com.facebook.presto.jdbc.PrestoDriver"}
```
[the remark this answers is not in the document]
Comment (kokosing) [reply presto-product-tests/conf/docker/common/compose-commons.sh]: No. One would like to use different version of presto-jdbc.

File: presto-product-tests/conf/docker/common/kerberos.yml [hunk 4]
```diff
@@ -20,3 +20,4 @@ services:
     image: 'prestodb/hdp2.5-hive-kerberized:${DOCKER_IMAGES_VERSION}'
     volumes:
       - ../../../conf/tempto/tempto-configuration-for-docker-kerberos.yaml:/docker/volumes/tempto/tempto-configuration-local.yaml
+    command: bash -c 'echo please provide command to application-runner >&2; exit 1'
```
[the remark this answers is not in the document]
Comment (kokosing) [reply presto-product-tests/conf/docker/common/kerberos.yml]: other application-runners are using different image. But I will change that thanks.

File: presto-product-tests/conf/docker/multinode-tls/docker-compose.yml (hunk 1 above)
Comment (kokosing) [reply presto-product-tests/conf/docker/multinode-tls/docker-compose.yml]: I have to admin that it takes time to get used to it.

Where possible "composition" is used, see how `conf/docker/common/jdbc_db.yml` is used. 

The general problem in `docker-compose` is that you often want to almost the same configuration like other configuration (e.g. singlenode), but you just want to change couple things. That way "inheritance" is much easier to do, easier to maintain, no copy-paste, hard to follow, easy to break.

This is why I recently added that whole configuration is displayed during the execution, so you exactly know what you are running.

File: presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml (hunk 2 above)
Comment (kokosing) [reply presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml]: There is a README file, how to use it.

I was thinking if it is a good idea to add them here, but I thought they may wors a documentation with real and working information how to configure simba jdbc with kerberos or ldap.

Though, I do not have strong opinion about this. I could add them to our infra instead. What do you think?

**Comment by kokosing** [comment]
@losipiuk comments addressed and rebased

File: presto-product-tests/conf/docker/multinode-tls/docker-compose.yml (hunk 1 above)
Comment (losipiuk) [reply presto-product-tests/conf/docker/multinode-tls/docker-compose.yml]: > That way "inheritance" is much easier to do, easier to maintain, no copy-paste, hard to follow, easy to break.

Yeah. I am fine with multiple `.yml` files which are used for composing the final one.  But maybe for inheritance (like overriding `standard.yml` with `kerberos.yml`) we could express that inheritance in `kerberos.yml` file itself.
And then use just `kerberos.yml` (not both `standard.yml` and `kerberos.yml`) in relevant profile specific docker-compose.sh? 
Expressing inheritance within `.yml` file can be done using `extension mechanism` from what I see (https://docs.docker.com/compose/extends/#extending-services)

> This is why I recently added that whole configuration is displayed during the execution, so you exactly know what you are running

Oh - I saw PR for Tempto config. Is there for docker compose also?

File: presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml (hunk 2 above)
Comment (losipiuk) [reply presto-product-tests/conf/tempto/tempto-configuration-for-docker-kerberos-simba.yaml]: I think it is nice to have them here. What I was suggesting, was that rather than removing those files, you could add more context related to them. A focused documentation explaining where to take driver from, and quoting specific commands which needs to be run to make use of those test-profiles, would be beneficial.

I think what we should opt for, is that person with some project knowledge, but not necessarily expert in the specific matter (product tests in this case) should be able to perform documented task (run product tests with simba drivers in this case).

PS. I am not suggesting that my PR fulfill that expectations very well, but yet I believe they should ;)

**Comment by losipiuk** [comment]
Product tests failed. Is that related?

File: presto-product-tests/conf/docker/multinode-tls/docker-compose.yml (hunk 1 above)
Comment (kokosing) [reply presto-product-tests/conf/docker/multinode-tls/docker-compose.yml]: > Expressing inheritance within .yml file can be done using extension mechanism

Good idea, I will give a try but in separate PR. A lot have changed around docker since we introduced `run_on_docker` product tests, maybe it is a right to time to refresh this.

> Oh - I saw PR for Tempto config. Is there for docker compose also?

https://github.com/prestodb/presto/commit/237154e950c28d0114dfa71831d3ae3670f11f71

Comment (kokosing) [reply presto-product-tests/conf/docker/multinode-tls/docker-compose.yml]: > Expressing inheritance within .yml file can be done using extension mechanism from what I see (https://docs.docker.com/compose/extends/#extending-services)

They suggest that there are two ways to reuse docker-compose files. One which we already following, and second which uses `extends`. The problem with `extends` is that it is not supported in `3.x` version of docker-compose. So it does not sound as a good idea to refactor to something which is going to be removed. ; /
